lib/common/path_collection.c

```diff
@@ -20,20 +20,30 @@
 
 #include "common/path_collection.h"
 
-int scion_path_collection_init(struct scion_path_collection **paths)
+static struct scion_path_collection *scion_path_collection_from_list(struct scion_linked_list *list)
 {
-	struct scion_path_collection *new_paths = malloc(sizeof(*paths));
-	if (new_paths == NULL) {
-		return SCION_MEM_ALLOC_FAIL;
+	struct scion_path_collection *paths = malloc(sizeof(*paths));
```

Comment:
I'm a bit unsure about this.
According to this: https://stackoverflow.com/questions/2022335/whats-the-point-of-malloc0
`malloc(0)` may return a non-NULL address.

Reply:
`sizeof` returns the size of the `struct scion_path_collection` here, not the amount of elements in the the path collection

Reply:
Right, ok, so what was wrong with returning `SCION_MEM_ALLOC_FAIL`?

Reply:
Actually, I don't even need this helper function anymore. I can move the code directly into the `scion_path_collection_init`.